aws/docker/worker/generate_profile_plot.py

```diff
@@ -35,105 +37,167 @@
-    meshgrid of the timestamps for each profile
-    :param netCDF4.Dataset nc: An open netCDF file
-    :param numpy.ndarray x: An array of time values
-    :param numpy.ndarray z: A multi-dimensional array representing the profiles
+    :param dataset_id: the deployment name example:'ce_311-20200708T1723'
+    :return: pandas DataFrame with deployment variable values
     '''
-    xv = np.ones(z.shape, dtype=np.object)
-    # NEED TO FIX ONCE netcdftime becomes a layer
-    # converter = utime(nc.variables['time'].units)
-    for i, timestamp in enumerate(np.squeeze(x)):
-        # xv[i, :] = converter.num2date(timestamp)
-        xv[i, :] = datetime.utcfromtimestamp(timestamp)
-
-    xv = mdates.date2num(xv)
-    return xv
-
-
-def get_plot(x, y, z, cmap, title='Glider Profiles', ylabel='Pressure (dbar)', zlabel='Temperature'):
+    e = ERDDAP(
+      server='http://gliders.ioos.us/erddap',
+      protocol='tabledap',
+    )
+    e.response = 'csv'
+    e.dataset_id = dataset_id
+    e.variables = [
+                    'depth',
+                    'latitude',
+                    'longitude',
+                    'salinity',
+                    'temperature',
+                    'conductivity',
+                    'density',
+                    'time',
+                   ]
+    df = e.to_pandas()
+    return df
+
+
+def get_variables(dataset, parameter):
+    '''
+    :param dataset: pandas DataFrame with deployment data
+    :param parameter: name of parameter to plot
+    :return x: list of datetime object, time
+            y, z: numpy.ndarray depth, values
+            x_name y_name z_name: string parameter name
+    '''
+    if 'Error' in dataset.keys()[0]:
+        print(dataset[dataset.keys()[0]][1])
+        exit()
+
+    # get the names of the columns from the dataset
+    y_name = [name for name in dataset.keys() if 'depth' in name]
+    x_name = [name for name in dataset.keys() if 'time' in name]
+    z_name = [name for name in dataset.keys() if parameter in name]
+
+    # get the data arrays
+    y = dataset[y_name[0]].values
+    z = dataset[z_name[0]].values
+    x = dataset[x_name[0]].values
+
+    # create list of datetime object from the numpy.ndarray of
+    # date/time string values in the format '%Y-%m-%dT%H:%M:%SZ'
+    # get the indices of non-nan timestamps
+    xv = list()
+    ii = list()
+    for i, d in enumerate(x):
+        try:
+            xv.append(datetime.strptime(d, '%Y-%m-%dT%H:%M:%SZ'))
+            ii.append(i)
+        except TypeError:
+            error = 'strptime() argument 1 must be str, not float'
+    #         print(error)
+
+    # select only non-nan values
+    yv = [y[n] for n in ii]
+    xv = [x[n] for n in ii]
+    zv = [z[n] for n in ii]
+
+    # convert arrays to datatime or float to mask invalid values in the next step
+    xv = np.array(xv, dtype='datetime64')
+    yv = np.array(yv, dtype='float')
+    zv = np.array(zv, dtype='float')
```

Comment:
Is there a more efficient way to do this? If I'm following correctly, you're creating numpy arrays for x, y z, where there are non Nan time indices? 

Can you do something like this:

```
    # get the data arrays
    y = dataset[y_name[0]].values
    z = dataset[z_name[0]].values
    x = dataset[x_name[0]].values
   
    xv = x[~np.isnan(x)]
    yv = y[~np.isnan(x)]
    zv = z[~np.isnan(x)]
```
From there you can cast/convert to the right type (datetime/floats).

Reply:
@leilabbb for the sake of timeliness in getting this update in, I'll merge and we can handle efficiency issues (and new variables) in a separate PR